lib/Common/Memory/RecyclerPointers.h

```diff
@@ -109,23 +135,33 @@ void WriteBarrier(T * address, size_t count)
 // Copy array content. Triggers write barrier on the dst array content if if
 // Allocator and element type determines write barrier is needed.
 //
-template <class Allocator, class T, class PolicyType = T>
-void CopyArray(T* dst, size_t dstCount, const T* src, size_t srcCount)
-{
-    js_memcpy_s(dst, sizeof(T) * dstCount, src, sizeof(T) * srcCount);
-    WriteBarrier<T, Allocator, PolicyType>(dst, dstCount);
-}
-template <class Allocator, class T, class PolicyType = T>
-void CopyArray(NoWriteBarrierPtr<T>& dst, size_t dstCount, const NoWriteBarrierPtr<T>& src, size_t srcCount)
+template <typename Allocator, typename T, typename PolicyType = T>
+struct WriteBarrierUtil
 {
-    return CopyArray<Allocator, T, PolicyType>((T*)dst, dstCount, (const T*)src, srcCount);
-}
-template <class Allocator, class T, class PolicyType = T>
-void CopyArray(WriteBarrierPtr<T>& dst, size_t dstCount, const WriteBarrierPtr<T>& src, size_t srcCount)
-{
-    return CopyArray<Allocator, T, PolicyType>((T*)dst, dstCount, (const T*)src, srcCount);
-}
+    static void CopyArray(T* dst, size_t dstCount, T*const src, size_t srcCount)
+    {
+        js_memcpy_s(dst, sizeof(T) * dstCount, src, sizeof(T) * srcCount);
+        WriteBarrier<T, Allocator, PolicyType>(dst, dstCount);
+    }
+    static void CopyArray(NoWriteBarrierPtr<T>& dst, size_t dstCount, const NoWriteBarrierPtr<T>& src, size_t srcCount)
+    {
+        return CopyArray<Allocator, T, PolicyType>((T*)dst, dstCount, (const T*)src, srcCount);
+    }
+    static void CopyArray(WriteBarrierPtr<T>& dst, size_t dstCount, const WriteBarrierPtr<T>& src, size_t srcCount)
+    {
+        return CopyArray<Allocator, T, PolicyType>((T*)dst, dstCount, (const T*)src, srcCount);
+    }
+};
 
+template <class T, class PolicyType>
+struct WriteBarrierUtil<Recycler, T, PolicyType>
```

Comment:
because function template can't be partial specialized

---
In reply to: [86643736](https://github.com/Microsoft/ChakraCore/pull/1920#discussion_r86643736) [](ancestors = 86643736)